src/localize.js

```diff
@@ -125,20 +125,58 @@ const strategies = {
   },
 };
 
+// Reference for ratios https://www.w3.org/International/articles/article-text-size
+const expansionRatio = {
+  10: 2.5,
+  20: 1.9,
+  30: 1.7,
+  40: 1.5,
+  50: 1.5,
+  60: 1.6,
+  70: 1.6,
+  over70: 1.3,
+};
+
+const isAlpha = character => {
+  return /[a-zA-Z]/.test(character);
+};
+
+const textExpander = (string, numDuplicateChars = 0, expansionRatio = 1) => {
+  let expandedText = '';
+  let count = 0;
+  const repeatTimes = Math.ceil(expansionRatio);
+
+  for (const c of string) {
+    if (isAlpha(c) && count < numDuplicateChars) {
+      expandedText += c.repeat(repeatTimes);
+      count += repeatTimes;
+    } else {
+      expandedText += c;
+    }
+  }
+  return expandedText;
+};
+
+const textExpansionHandler = (string, toBeExpanded) => {
+  const strLen = string.length;
+  const ratioKey = Math.ceil(strLen / 10) * 10;
+  const ratio =
+    ratioKey > 70 ? expansionRatio.over70 : expansionRatio[ratioKey];
+  const numDuplicateChars = Math.ceil(strLen * ratio) - strLen;
```

Comment:
What if the string is very short? For example `hi`.

`Math.ceil(2*2.5)-2 = 5-2 = 3`  indicates that 3 characters need to be duplicated but the string only contains 2 characters.

Reply:
This is addressed by the
` const repeatTimes = Math.ceil(expansionRatio);` in the `textExpander` method.
repeatTimes address this issue by increasing the number of duplicates. Hence, in the case of hi, it would duplicate each character 3 times getting a 300% total size of the string, which will be "hhhiii". This is accepted by the IBM standards that accept 200%-300% for strings < 10 in length. And the for loop will break once the characters in the string have ended. so no logical errors will occur, to the extend of my knowledge.

However, I understand how this can confuse others. we can use an a ternary operator where `const numDuplicateChars =  (Math.ceil(strLen * ratio) - strLen)> strLen ? strLen :  (Math.ceil(strLen * ratio) - strLen)`. This will make it set numDuplicateChars  to 2 in the described case and any case where the number of duplicate characters is longer than the word. What do you think ?